internal/broadcaster/utxo_creator.go

```diff
@@ -7,97 +7,113 @@ import (
-
 	"github.com/bitcoin-sv/arc/internal/metamorph/metamorph_api"
 	"github.com/bitcoin-sv/arc/pkg/keyset"
+	sdkTx "github.com/bitcoin-sv/go-sdk/transaction"
 )
 
 type UTXOCreator struct {
 	Broadcaster
-	keySets []*keyset.KeySet
+	keySet *keyset.KeySet
+	wg     sync.WaitGroup
+}
+type splittingOutput struct {
+	satoshis uint64
+	vout     uint32
 }
 
-func NewUTXOCreator(logger *slog.Logger, client ArcClient, keySets []*keyset.KeySet, utxoClient UtxoClient, isTestnet bool, opts ...func(p *Broadcaster)) (*UTXOCreator, error) {
+func NewUTXOCreator(logger *slog.Logger, client ArcClient, keySet *keyset.KeySet, utxoClient UtxoClient, isTestnet bool, opts ...func(p *Broadcaster)) (*UTXOCreator, error) {
 	b, err := NewBroadcaster(logger, client, utxoClient, isTestnet, opts...)
 	if err != nil {
 		return nil, err
 	}
 
 	creator := &UTXOCreator{
 		Broadcaster: b,
-		keySets:     keySets,
+		keySet:      keySet,
 	}
 
 	return creator, nil
 }
 
-func (b *UTXOCreator) CreateUtxos(requestedOutputs int, requestedSatoshisPerOutput uint64) error {
-	for _, ks := range b.keySets {
-		b.logger.Info("creating utxos", slog.String("address", ks.Address(!b.isTestnet)))
+func (b *UTXOCreator) Wait() {
+	b.wg.Wait()
+}
+func (b *UTXOCreator) Start(requestedOutputs int, requestedSatoshisPerOutput uint64) error {
+	// Increment the WaitGroup counter
+	b.wg.Add(1)
+
+	// Use a goroutine for concurrent execution
+	go func() {
+		defer func() {
+			// Log the shutdown message
+			b.logger.Info("shutting down")
+			// Mark this goroutine as done
+			b.wg.Done()
+		}()
+
+		b.logger.Info("creating utxos", slog.String("address", b.keySet.Address(!b.isTestnet)))
 
 		requestedOutputsSatoshis := int64(requestedOutputs) * int64(requestedSatoshisPerOutput)
 
-		confirmed, unconfirmed, err := b.utxoClient.GetBalanceWithRetries(b.ctx, ks.Address(!b.isTestnet), 1*time.Second, 5)
+		confirmed, unconfirmed, err := b.utxoClient.GetBalanceWithRetries(b.ctx, b.keySet.Address(!b.isTestnet), 1*time.Second, 5)
 		if err != nil {
-			return err
+			b.logger.Error("failed to get balance", slog.String("err", err.Error()))
+			return
 		}
 
 		balance := confirmed + unconfirmed
 
 		if requestedOutputsSatoshis > balance {
-			return fmt.Errorf("requested total of satoshis %d exceeds balance on funding keyset %d", requestedOutputsSatoshis, balance)
+			b.logger.Error("requested total exceeds balance", slog.Int64("requested", requestedOutputsSatoshis), slog.Int64("balance", balance))
+			return
 		}
 
-		utxos, err := b.utxoClient.GetUTXOsWithRetries(b.ctx, ks.Script, ks.Address(!b.isTestnet), 1*time.Second, 5)
+		utxos, err := b.utxoClient.GetUTXOsWithRetries(b.ctx, b.keySet.Script, b.keySet.Address(!b.isTestnet), 1*time.Second, 5)
 		if err != nil {
-			return err
+			b.logger.Error("failed to get utxos", slog.String("err", err.Error()))
+			return
 		}
 
 		utxoSet := list.New()
 		for _, utxo := range utxos {
-			// collect right sized utxos
```

Comment:
We shouldn't remove such comments as they can facilitate the understanding of the code